packages/embark/src/lib/modules/ens/index.js

```diff
@@ -366,23 +366,29 @@ class ENS {
   }
 
   addENSToEmbarkJS() {
-    const self = this;
-
-    // get namehash, import it into file
-    self.events.request("version:get:eth-ens-namehash", function (EnsNamehashVersion) {
-      let currentEnsNamehashVersion = require('../../../../package.json').dependencies["eth-ens-namehash"];
-      if (EnsNamehashVersion !== currentEnsNamehashVersion) {
-        self.events.request("version:getPackageLocation", "eth-ens-namehash", EnsNamehashVersion, function (err, location) {
-          self.embark.registerImportFile("eth-ens-namehash", self.fs.dappPath(location));
-        });
+    this.events.request('version:downloadIfNeeded', 'eth-ens-namehash', (err, location) => {
+      if (err) {
+        this.logger.error(__('Error downloading NameHash'));
+        return this.logger.error(err.message || err);
       }
-    });
 
-    let code = self.fs.readFileSync(utils.joinPath(__dirname, 'ENSFunctions.js')).toString();
-    code += "\n" + self.fs.readFileSync(utils.joinPath(__dirname, 'embarkjs.js')).toString();
-    code += "\nEmbarkJS.Names.registerProvider('ens', __embarkENS);";
+      this.events.once('code-generator:ready', () => {
+        this.events.request('code-generator:symlink:generate', location, 'eth-ens-namehash', (err, symlinkDest) => {
```

Comment:
Are the symlink always created?

Reply:
Yes, should be. Even when the Dapp uses Embark's own pipeline. Is there a problem?

Reply:
I just wonder if we should do it all time, this is only usefull for CRA, it might not be needed for Vue or  Angular.
What do you think?

Reply:
Hard to know until we tested. Also, I'm not a fan of putting conditions for specific pipelines, especially since we're supposed to be agnostic.

Reply:
>  I'm not a fan of putting conditions for specific pipelines

Totally agree with that
This is exactly the point behind my question, is creating those symlinks not related to CRA only.

I would have imagined the symlink to be in the template dedicated to CRA

Reply:
Ah ok, I understand your point. I don't know yet for the other pipelines, but also, it's gonna be hard to create a template that tells Embark to create the symlinks, even with a plugin, because it would need to NOT do this part and do the one in the plugin instead.

Anyway, I feel like the symlinks are more a safeguard to make sure all pipelines work over a CRA fix.

Reply:
Fine by me, also up to you, you know better than me